### PR TITLE
[ENG-1392] Backwards compatable Django upgrade to 2.0.13 

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose

Get backwards compatible code out of the way before we flip over to Django 2

## Changes
- Make Many-to-manys use direct assignment
- add  `on_delete` kwarg to  foreign key fields
- Get urls helpers from  `django.urls` instead of `django.core.urlresolvers` which is depreciated
- Change url namespaces for new Django 2 url format.

## QA Notes

- No data migration
- Low risk
- No permissions code touched
- No user facing changes

## Documentation

Tech task, no new docs.

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/ENG-1392

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -62,7 +62,7 @@ requests==2.20.0
 urllib3==1.22
 oauthlib==2.0.6
 requests-oauthlib==0.8.0
-raven==5.32.0
+raven==6.4.0
 
 # API requirements
 Django==1.11.15  # pyup: <2.0 # Remove this when we're on Py3
```
Comment (brianjgeiger) [inline requirements.txt]: Drive-by: Might be good to upgrade us to the last version of 1.11 while you're here.

File: requirements.txt [hunk 2]
```diff
@@ -62,10 +62,10 @@ requests==2.20.0
 urllib3==1.22
 oauthlib==2.0.6
 requests-oauthlib==0.8.0
-raven==5.32.0
+raven==6.4.0
 
 # API requirements
-Django==1.11.15  # pyup: <2.0 # Remove this when we're on Py3
+Django==1.11.28  # pyup: <2.0 # Remove this when we're on Py3
```
Comment (corbinSanders) [inline requirements.txt]: Should we go ahead and remove this comment since we're on py3?

**Comment by corbinSanders** [comment]
Looks good besides my one comment!

File: osf/models/collection.py [hunk 3]
```diff
@@ -181,8 +181,18 @@ def save(self, *args, **kwargs):
 
         if first_save:
             # Set defaults for M2M
-            self.collected_types = ContentType.objects.filter(app_label='osf', model__in=['abstractnode', 'collection', 'preprint'])
-            # Set up initial permissions
+            self.collected_types = ContentType.objects.filter(
+                app_label='osf',
+                model__in=['abstractnode', 'collection', 'preprint']
+            )
+            content_type = ContentType.objects.filter(
```
Comment (brianjgeiger) [inline osf/models/collection.py]: Out of curiosity, what changed that caused this to be added in?

Comment (Johnetordoff) [reply osf/models/collection.py]: just a copy and paste error
